data/lua/cave_map_generator.lua

```diff
@@ -123,7 +122,7 @@ function callbacks.generate_map(params)
 			local laziness = math.max(v.data.laziness or 1, 1)
 			local width = math.max(v.data.width or 1, 1)
 			local jagged = v.data.jagged or 0
-			local calc = function(x, y, current_cost)
+			local calc = function(x, y)
```

Comment:
Thinking about it, I have a feeling that removing the parameter might actually give an error message. This function will be _called_ with three parameters (from the C++ implementation of `wesnoth.find_path), so if I understand correctly, it needs to also be _declared_ to accept three parameters. (Using `...` would technically work too.)

I'm not 100% sure on this point, but it's something that should be checked, at least.

Reply:
@CelticMinstrel 
https://www.lua.org/pil/5.html
> You can call a function with a number of arguments different from its number of parameters. Lua adjusts the number of arguments to the number of parameters, as it does in a multiple assignment: Extra arguments are thrown away; extra parameters get nil. For instance,...

Reply:
Ah, I see. In that case, I guess it's fine.